src/markers/VMarker.vue

```diff
@@ -50,35 +50,23 @@
         required: false,
       },
     },
-    setup(props, { emit }: SetupContext) {
+    setup(props, { emit, slots }: SetupContext) {
       let map = injectStrict(MapKey);
       let marker: Marker = new Marker(props.options);
-      let loaded: Ref<boolean> = ref(true);
-
-      map.value.on('style.load', () => {
-        // https://github.com/mapbox/mapbox-gl-js/issues/2268#issuecomment-401979967
-        const styleTimeout = () => {
-          if (!map.value.isStyleLoaded()) {
-            loaded.value = false;
-            setTimeout(styleTimeout, 200);
-          } else {
-            loaded.value = true;
-          }
-        };
-        styleTimeout();
-      });
```

Comment:
If you've added the `loaded` check in map, maybe we can refactor this code from all the other files ?

Reply:
based on the issue that is linked it seems that the style can be changed and then the map.value.isStyleLoaded() can return false even when the map was loaded once before. I think leaving this as is in VMarker and other components is safer then dooing it only in VMap component when load event is fired from mapbox.

The use case for this code is when you change the style of the map and then add new marker.

Reply:
@Buk1m Do you think it would be ok to reload all child components (markers, …) centrally from map when style has changed?

Because this would allow simplify components.

Reply:
@vinayakkulkarni Could you please tell if I can solve style change by children reload? It will be small performance degradation on style change, but huge simplification of child components.

Thank you for decision. I am going to work on another issues and I will need to merge this, because otherwise I will be forced to commit everything just to my fork and not care about PRs. It is hard to advocate open source contributions to my clients when it will delay fixes and make it more expensive.

Reply:
Hi @iBobik, sorry for the late message I must have missed the previous notification. 

I only worked with mapbox custom styles in a way that styles were loaded at the beginning and never changed, in which case your solution would work. (Un)fortunately  you can change them after the map was loaded simply by calling [map.setStyle](https://docs.mapbox.com/mapbox-gl-js/example/setstyle/) and then `map.value.isStyleLoaded()` will return false again.

Loading new styles can take a while as it is usually a http request. So if the styles were changed and you will try to add new marker before loading is finished, then you will encounter "Style not done loading" error (At least according to the issue from comment).

There is no way to listen if styles were changed so you can either wrap setStyles method to know when the styles were changed and delay map manipulations until `map.value.isStyleLoaded()` return true or stick with the current solution. 

So the case is:
1. Map has fully loaded once
2. Styles are changed
3. New marker is added before new styles are fully loaded.
4. Possible "Style not done loading" error 

Keep in mind that the issue is old and I didn't test this exact case myself.

Reply:
Do you have some idea how to solve it without having this timeout logic in every component?

Is it enough to have own method for style switching what will unload all children for the time of loading? 

I am looking for solution what will simplify v-mapbox' codebase, because we are not able to do everything perfectly. In this project's status we have difficulties to push it to the stable state just for basic use-cases, so I think we should not paralyse development by requiring to cover too much use-cases.